pkg/roachprod/vm/azure/utils.go

```diff
@@ -33,7 +34,10 @@ const azureStartupTemplate = `#!/bin/bash
 set -xe
 mount_opts="defaults"
 
-{{if .AttachedDiskLun}}
+{{if .DiskControllerNVMe}}
+# Setup nvme network storage, need to remove nvme OS disk from the device list.
+devices=($(realpath -qe /dev/disk/by-id/nvme-* | grep -v "nvme0n1" | sort -u))
+{{else if .AttachedDiskLun}}
```

Comment:
The `else` block handles a condition that should never happen in practice, right? The "setup local storage" part. Maybe we can remove that logic to make it less confusing.